cuqipy_pytorch/distribution.py

```diff
@@ -395,3 +395,109 @@ def gradient(self, v1, v2=None):
+        return self._location
+
+    @location.setter
+    def location(self, value):
+        self._location = value
+        self._set_dist()
+    
+    @property
+    def scale(self):
+        return self._scale
+
+    @scale.setter
+    def scale(self, value):
+        self._scale = value
+        self._set_dist()
+
+    def _set_dist(self):
+        """ Set the pytorch distribution if both values are specified """
+        if hasattr(self, '_location') and hasattr(self, '_scale'):
+
+            # Define location value
+            loc = self.location
+            if isinstance(loc, numbers.Number):
+                loc = loc*torch.ones(self.dim)
+            if isinstance(loc, np.ndarray):
+                loc = torch.tensor(loc)
+
+            # Define scale value
+            scale = self._scale
+            if isinstance(scale, numbers.Number):
+                scale = scale*torch.ones(self.dim)
+            if isinstance(scale, np.ndarray):
+                scale = torch.tensor(scale)
+
+            # If both are tensors we create dist
+            if torch.is_tensor(loc) and torch.is_tensor(scale):
+                self._dist = torch.distributions.Laplace(loc, scale)
+
+    def logpdf(self, value):
+        return torch.sum(self._dist.log_prob(value))
+
+    def _sample(self, n):
+        return self._dist.sample(torch.Size((n,)))
+
+class Cauchy(Distribution):
+
+    def __init__(self, location, scale, is_symmetric=True, **kwargs):
+        super().__init__(is_symmetric=is_symmetric, **kwargs) 
+        self.location = location
+        self.scale = scale
+
+    @property
+    def location(self):
+        return self._location
+
+    @location.setter
+    def location(self, value):
+        self._location = value
+        self._set_dist()
+    
+    @property
+    def scale(self):
+        return self._scale
+
+    @scale.setter
+    def scale(self, value):
+        self._scale = value
+        self._set_dist()
+
+    def _set_dist(self):
+        """ Set the pytorch distribution if both values are specified """
+        if hasattr(self, '_location') and hasattr(self, '_scale'):
+
+            # Define location value
+            loc = self.location
+            if isinstance(loc, numbers.Number):
+                loc = loc*torch.ones(self.dim)
+            if isinstance(loc, np.ndarray):
+                loc = torch.tensor(loc)
+
+            # Define scale value
+            scale = self._scale
+            if isinstance(scale, numbers.Number):
+                scale = scale*torch.ones(self.dim)
+            if isinstance(scale, np.ndarray):
+                scale = torch.tensor(scale)
+
+            # If both are tensors we create dist
+            if torch.is_tensor(loc) and torch.is_tensor(scale):
+                self._dist = torch.distributions.Cauchy(loc, scale)
+
+    def logpdf(self, value):
+        return torch.sum(self._dist.log_prob(value))
+
+    def _sample(self, n):
+        return self._dist.sample(torch.Size((n,)))
```

Comment:
May be an issue can be added to address code redundancy in some distribution wrappers to be considered in the future.

Reply:
#14

Reply:
Great!